unittests/unit/test_ConstraintSolver.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2011-2019, The DART development contributors
+ * All rights reserved.
+ *
+ * The list of contributors can be found at:
+ *   https://github.com/dartsim/dart/blob/master/LICENSE
+ *
+ * This file is provided under the following "BSD-style" License:
+ *   Redistribution and use in source and binary forms, with or
+ *   without modification, are permitted provided that the following
+ *   conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ *   CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *   INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ *   MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *   DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ *   CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *   LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+ *   USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ *   AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *   LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *   ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *   POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <iostream>
+
+#include <gtest/gtest.h>
+
+#include "dart/constraint/ConstraintBase.hpp"
+#include "dart/constraint/ConstraintSolver.hpp"
+#include "dart/simulation/World.hpp"
+
+#include "TestHelpers.hpp"
+
+using namespace dart;
+using namespace dart::neural;
+
+TEST(ConstraintSolver, SIMPLE)
+{
+  std::shared_ptr<simulation::World> world
+      = std::make_shared<simulation::World>();
+  auto solver = world->getConstraintSolver();
+
+  for (auto constraintGroup : solver->getConstrainedGroups())
+  {
+    std::vector<s_t*> impulses
+        = solver->solveConstrainedGroup(constraintGroup, world.get());
+    solver->applyConstraintImpulses(constraintGroup.getConstraints(), impulses);
+  }
```

Comment:
Just to make it possible for this test to fail, maybe we add a check down here to see if the velocity updates are what we expect. If you set up the `world` object as being two objects colliding, then by the time you reach this point, their relative velocity should be >= 0, so you can calculate `relativeVelocity` and add an `EXPECT_TRUE(relativeVelocity >= 0)` or something, which makes it a real unit test!